File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/client/createClientRequest.ts
import { Client } from "../../../..";

export interface CreateClientRequest {
    name: string;
    middleName: string;
    lastName: string;
    dateOfBirth: Date;
    address: string;
    phone: string;
    email: string;
}

export function mapClientToCreateClientRequest(client: Client): CreateClientRequest {
    return {
        name: client.name,
        middleName: client.middleName,
        lastName: client.lastName,
        dateOfBirth: client.dateOfBirth,
        address: client.address,
        phone: client.phone,
        email: client.email
    }
}

export function getDefaultCreateClientRequest(): CreateClientRequest {
    return {
        name: 'Jane',
        middleName: 'M',
        lastName: 'Doe',
        address: '456 Main St',
        phone: '0987654321',
        email: 'jane.doe@example.com',
        dateOfBirth: new Date('1992-02-02')
    }
}

File: src/ELibrary.Backend/ShopApi/Features/ClientFeature/Dtos/CreateClientRequest.cs
﻿namespace ShopApi.Features.ClientFeature.Dtos
{
    public class CreateClientRequest
    {
        public string? Name { get; set; }
        public string? MiddleName { get; set; }
        public string? LastName { get; set; }
        public DateTime DateOfBirth { get; set; }
        public string? Address { get; set; }
        public string? Phone { get; set; }
        public string? Email { get; set; }
    }
}
